kv/kvserver: TestRaftForceCampaignPreVoteCheckQuorum failed

kv/kvserver.TestRaftForceCampaignPreVoteCheckQuorum [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20391364?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20391364?buildTab=artifacts#/) on master @ [f276f1ade7253ccbb6f3073ce215ca464b7d8220](https://github.com/cockroachdb/cockroach/commits/f276f1ade7253ccbb6f3073ce215ca464b7d8220):


```
=== RUN   TestRaftForceCampaignPreVoteCheckQuorum
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/15f2e1706383de9d6af3292f657b4304/logTestRaftForceCampaignPreVoteCheckQuorum3093547704
    test_log_scope.go:82: use -show-logs to present logs inline
    client_raft_test.go:6183: n1 StateLeader at term=6 commit=20
    client_raft_test.go:6184: n1 is leader in term 6
    client_raft_test.go:6189: n3 campaigning
    client_raft_test.go:6195: n1 StateLeader at term=6 commit=20
    client_raft_test.go:6195: n1 StateLeader at term=6 commit=21
    client_raft_test.go:6195: n1 StateLeader at term=6 commit=21
    client_raft_test.go:6195: n1 StateLeader at term=6 commit=21
    client_raft_test.go:6195: n1 StateLeader at term=6 commit=21
    client_raft_test.go:6195: n1 StateLeader at term=6 commit=22
    client_raft_test.go:6195: n1 StateLeader at term=6 commit=22
    client_raft_test.go:6195: n1 StateLeader at term=6 commit=22
    client_raft_test.go:6195: n1 StateLeader at term=6 commit=22
    client_raft_test.go:6195: n1 StateLeader at term=6 commit=22
    client_raft_test.go:6195: n1 StateLeader at term=6 commit=22
    client_raft_test.go:6195: n1 StateLeader at term=6 commit=22
    client_raft_test.go:6195: n1 StateLeader at term=6 commit=23
    client_raft_test.go:6195: n1 StateLeader at term=6 commit=23
    client_raft_test.go:6195: n1 StateLeader at term=6 commit=23
    client_raft_test.go:6195: n1 StateLeader at term=6 commit=23
    client_raft_test.go:6195: n1 StateLeader at term=6 commit=23
    client_raft_test.go:6195: n1 StateLeader at term=6 commit=23
    client_raft_test.go:6195: n1 StateLeader at term=6 commit=24
    client_raft_test.go:6192: 
        	Error Trace:	pkg/kv/kvserver/client_raft_test.go:6192
        	Error:      	Condition never satisfied
        	Test:       	TestRaftForceCampaignPreVoteCheckQuorum
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/15f2e1706383de9d6af3292f657b4304/logTestRaftForceCampaignPreVoteCheckQuorum3093547704
--- FAIL: TestRaftForceCampaignPreVoteCheckQuorum (12.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRaftForceCampaignPreVoteCheckQuorum.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-53912

--- a/pkg/kv/kvserver/client_raft_test.go
+++ b/pkg/kv/kvserver/client_raft_test.go
@@ -6165,6 +6165,9 @@ func TestRaftForceCampaignPreVoteCheckQuorum(t *testing.T) {
 	// Create a range, upreplicate it, and replicate a write.
 	key := tc.ScratchRange(t)
 	desc := tc.AddVotersOrFatal(t, key, tc.Targets(1, 2)...)
+	// Wait for the config changes corresponding to adding voters to be finalized.
+	// Otherwise, the forced campaign can fail and not retry. See #152555.
+	require.NoError(t, tc.WaitForVoters(key, tc.Targets(0, 1, 2)...))
 	_, pErr := kv.SendWrapped(ctx, sender, incrementArgs(key, 1))
 	require.NoError(t, pErr.GoError())
 	tc.WaitForValues(t, key, []int64{1, 1, 1})

